File: backend/src/main/java/com/huiwutong/conference/entity/Ticket.java
package com.huiwutong.conference.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 票务实体
 */
@Data
@TableName("tickets")
public class Ticket implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @TableField("name")
    private String name;

    @TableField("price")
    private BigDecimal price;

    @TableField("description")
    private String description;

    @TableField("stock")
    private Integer stock;

    @TableField("sold_count")
    private Integer soldCount;

    @TableField("status")
    private Integer status;

    @TableField("ticket_type")
    private Integer ticketType;

    @TableField("min_persons")
    private Integer minPersons;

    @TableField("max_persons")
    private Integer maxPersons;

    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/admin/src/api/ticket.ts
import request from '@/utils/request'
import type { PageData } from '@/utils/request'

export interface Ticket {
  id: number
  name: string
  price: number
  description: string
  stock: number
  soldCount: number
  status: number
  ticketType: number   // 1=个人票 2=团体票
  minPersons: number | null
  maxPersons: number | null
  createdAt: string
  updatedAt: string
}

export interface TicketForm {
  name: string
  price: number | string
  stock: number | string
  description: string
  ticketType: number   // 1=个人票 2=团体票
  minPersons?: number | null
  maxPersons?: number | null
}

export interface TicketListParams {
  page?: number
  pageSize?: number
  status?: number
}

export const ticketApi = {
  getList(params: TicketListParams) {
    return request.get<PageData<Ticket>, PageData<Ticket>>('/admin/ticket', { params })
  },

  getDetail(id: number) {
    return request.get<Ticket, Ticket>(`/admin/ticket/${id}`)
  },

  create(data: TicketForm) {
    return request.post<Ticket, Ticket>('/admin/ticket', data)
  },

  update(id: number, data: TicketForm) {
    return request.put<Ticket, Ticket>(`/admin/ticket/${id}`, data)
  },

  updateStatus(id: number, status: number) {
    return request.put(`/admin/ticket/${id}/status`, { status })
  },

  adjustStock(id: number, stock: number) {
    return request.put(`/admin/ticket/${id}/stock`, { stock })
  },

  delete(id: number) {
    return request.delete(`/admin/ticket/${id}`)
  }
}
